components/brave_service_keys/service_key_utils.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SERVICE_KEYS_SERVICE_KEY_UTILS_H_
+#define BRAVE_COMPONENTS_BRAVE_SERVICE_KEYS_SERVICE_KEY_UTILS_H_
+
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "url/gurl.h"
+
+namespace brave_service_keys {
+
+std::pair<std::string, std::string> GetDigestHeader(const std::string& payload);
```

Comment:
please add comments explaining what these methods do and in particular what they return

Reply:
also I'm not really a fan of `std::pair<std::string, std::string>` used all over the place here. It's very opaque. It's not a blocker for this I guess, but in a follow-up it would be better to use actual types here like structs or re-use existing structs like `HttpRequestHeaderKeyValuePair`